Tests/Tests/IncrementalStore.m

```diff
@@ -731,6 +734,144 @@ - (void)testFetchConstraints
+        req.propertiesToUpdate = @{ @"check" : @(NO) };
+        req.resultType = NSUpdatedObjectsCountResultType;
+        NSBatchUpdateResult *res = (NSBatchUpdateResult *)[moc executeRequest:req error:&err];
+
+        if (!CDTISSupportBatchUpdates) {
+            XCTAssertNil(res, @"Result should be nil since batch updates are not supported");
+            XCTAssertTrue([err.domain isEqualToString:CDTISErrorDomain],
+                          @"Error domain should indicate error source");
+            XCTAssertTrue(err.code == CDTISErrorExectueRequestTypeUnkown,
+                          @"Error code should identify error reason");
+        } else {
+            XCTAssertNotNil(res, @"Expected results: %@", err);
+            NSLog(@"%@ objects updated", res.result);
+
+            /**
+             *  Fetch checked entries
+             */
+            fr.predicate = checked;
+            results = [moc executeFetchRequest:fr error:&err];
+            XCTAssertNotNil(results, @"Expected results: %@", err);
+
+            XCTAssertTrue([results count] == (num_entries / 4),
+                          @"results count should be %d is %lu", (num_entries / 4),
+                          (unsigned long)[results count]);
+        }
+    }
+
+    /**
+     *  Batch update date, string, integer, and float attributes
+     *  Request objectIDs to be returned
+     */
+    {
+        NSBatchUpdateRequest *req = [[NSBatchUpdateRequest alloc] initWithEntityName:@"Entry"];
+        req.predicate = [NSPredicate predicateWithFormat:@"check == YES"];
+        NSDate *now = [NSDate date];
+        req.propertiesToUpdate = @{
+            @"created_at" : now,
+            @"text" : @"foobar",
+            @"i16" : @(32),
+            @"fpFloat" : @(M_PI_2),
+            @"fpDouble" : @(M_PI)
+        };
+        req.resultType = NSUpdatedObjectIDsResultType;
+        NSBatchUpdateResult *res = (NSBatchUpdateResult *)[moc executeRequest:req error:&err];
+
+        if (!CDTISSupportBatchUpdates) {
+            XCTAssertNil(res, @"Result should be nil since batch updates are not supported");
+            XCTAssertTrue([err.domain isEqualToString:CDTISErrorDomain],
+                          @"Error domain should indicate error source");
+            XCTAssertTrue(err.code == CDTISErrorExectueRequestTypeUnkown,
+                          @"Error code should identify error reason");
+        } else {
+            XCTAssertNotNil(res, @"Expected results: %@", err);
+
+            XCTAssertTrue([res.result count] == (num_entries / 4),
+                          @"results count should be %d is %lu", (num_entries / 4),
+                          (unsigned long)[res.result count]);
+
+            [res.result enumerateObjectsUsingBlock:^(NSManagedObjectID *objID, NSUInteger idx,
+                                                     BOOL *stop) {
+              Entry *e = (Entry *)[moc objectWithID:objID];
+              if (![e isFault]) {
```

Comment:
NACK: The pattern I'm using here is the one advocated in the bignerdranch article on batch updating:

https://www.bignerdranch.com/blog/new-in-core-data-and-ios-8-batch-updating/